### DISCUSSION TITLE
basic verification of `optional_packages_version`?

### DISCUSSION THREAD

**Discussion by wyli** [Project-MONAI/model-zoo#30] [Ideas]
Related to this ticket https://github.com/Project-MONAI/MONAI/issues/4456,
the metadata json should also include the info about pytorch-ignite version. For example this bundle clearly depends on ignite, but it's not described in the metadata: https://github.com/Project-MONAI/model-zoo/blob/0ded5a41d8fd5bfc43e1d0912e02877b5cc2d232/models/spleen_ct_segmentation/configs/metadata.json#L11-L13

I guess in general the question is how we make sure the stated `optional_packages_version` is indeed the minimally required set of packages to run a bundle.

**Comment by Nic-Ma** [comment]
Hi @wyli ,

Thanks for raising the question.
I asked this question several times during our previous discussions: How to find out the dependent packages in a bundle? Developers may use a docker to develop the bundle, don't know some deep dependencies at all.
CC @ericspod 

Thanks.

**Comment by ericspod** [comment]
Where an imported package is used isn't easy to test, we could create a virtual environment with a github action and install everything specified then try to instantiate every object in one of the scripts to see if it's possible, this would catch the lack of pytorch-ignite for example when instantiating any engine type. This would miss cases where the package is used in code that isn't run until train/inference time. A more in-depth method would be to run static code analysis to collect together all instances of package importation to get a list of all dependencies, this would omit only the most bizarre cases of `exec/eval` or `importlib` being used to import packages some other way.
